libraries/Microsoft.Bot.Schema/ActivityEx.cs

```diff
@@ -178,7 +178,36 @@ public static ITraceActivity CreateTraceActivity(string name, string valueType =
         /// <summary>
         /// True if the Activity is of the specified activity type
         /// </summary>
-        protected bool IsActivity(string activity) { return string.Compare(this.Type?.Split('/').First(), activity, true) == 0; }
+        protected bool IsActivity(string activityType)
```

Comment:
Could you add a detailed comment here - similar to what you have in the PR - that says:
"here's the easy way to do this: [simple code]"

"... but this is called all the time, and the profiler says its worth optimizing, so the code below is optimized to do this same check more efficiently."

Reply:
✔️